src/if.js

```diff
@@ -43,20 +43,42 @@ export class If {
   * @param newValue The new value
   */
   valueChanged(newValue) {
+    if (this.__queuedChanges) {      
+      this.__queuedChanges.push(newValue);
+      return;
+    }
+
+    let maybePromise = this._runValueChanged(newValue);
+    if (maybePromise instanceof Promise) {
+      let queuedChanges = this.__queuedChanges = [];
+
+      let runQueuedChanges = () => {
+        if (!queuedChanges.length) {
+          delete this.__queuedChanges;       
+          return;
+        }        
+        let nextPromise = this._runValueChanged(queuedChanges.shift()) || Promise.resolve();          
```

Comment:
I suspect, i'm missing smth really obvious, 
but right now it seems to me, that 
1) `let queuedChanges = this.__queuedChanges = [];` always assigns empty array to `queuedChanges`
2) therefore because `queuedChanges.length` is 0, then `!queuedChanges.length` is `true`
3) therefore it seems to me, that **this line is never reached**, as function returns from if block

Is this code written as intended - what am i missing?